app/src/main/java/org/simple/clinic/patient/PatientRepository.kt

```diff
@@ -49,10 +50,22 @@ class PatientRepository @Inject constructor(
   private var ongoingNewPatientEntry: OngoingNewPatientEntry = OngoingNewPatientEntry()
 
   fun search(name: String): Observable<List<PatientSearchResult>> {
+    val timingTracker = OperationTimingTracker("Search Patient", utcClock)
+
+    val fetchPatientNameAnalytics = "Fetch Name and Id"
+    val fuzzyFilterPatientNameAnalytics = "Fuzzy Filtering By Name"
+    val fetchPatientDetailsAnalytics = "Fetch Patient Details"
+    val sortByVisitedFacilityAnalytics = "Sort By Visited Facility"
+
     val patientUuidsMatchingName = database.patientSearchDao()
         .nameAndId(PatientStatus.ACTIVE)
         .toObservable()
-        .switchMapSingle { searchPatientByName.search(name, it) }
+        .switchMapSingle {
+          timingTracker.stop(fetchPatientNameAnalytics)
+          timingTracker.start(fuzzyFilterPatientNameAnalytics)
+          searchPatientByName.search(name, it)
+              .doOnSuccess { timingTracker.stop(fuzzyFilterPatientNameAnalytics) }
+        }
 
```

Comment:
Can this `doOnSuccess` be moved outside `switchMapSingle`? The indentation will look a bit cleaner.